src/clients/inst_crowbar_patterns.rb

```diff
@@ -28,12 +28,47 @@ class InstCrowbarPatternsClient < Client
     def main
 
       Yast.import "GetInstArgs"
+      Yast.import "Installation"
+      Yast.import "Kernel"
       Yast.import "PackagesProposal"
+      Yast.import "PackagesUI"
+      Yast.import "Pkg"
+      Yast.import "Stage"
 
       dialog_ret = :auto
       patterns = [ "cloud_admin" ]
 
-      unless GetInstArgs.going_back
+      return dialog_ret if GetInstArgs.going_back
+
+      if Stage.normal
+        # In normal stage, we need to use the functionality from
+        # inst_add-on_software to setup source and target
+
+        Pkg.TargetInit(Installation.destdir, false)
+        Pkg.SourceStartManager(true)
+        patterns.each do |pattern|
+          Pkg.ResolvableInstall(pattern, :pattern)
+        end
+
+        ret = PackagesUI.RunPatternSelector
+        Builtins.y2milestone("RunPatternSelector returned %1", ret)
+
+        dialog_ret = :next
+        dialog_ret = :abort if ret == :cancel
```

Comment:
NP: I'd join this to a single line: `dialog_ret =  (ret == :cancel) ? :abort : :next`